library/std/src/sys/unix/time.rs

```diff
@@ -20,17 +20,15 @@ impl Timespec {
 
     fn sub_timespec(&self, other: &Timespec) -> Result<Duration, Duration> {
         if self >= other {
-            Ok(if self.t.tv_nsec >= other.t.tv_nsec {
-                Duration::new(
-                    (self.t.tv_sec - other.t.tv_sec) as u64,
-                    (self.t.tv_nsec - other.t.tv_nsec) as u32,
-                )
+            let (secs, nsec) = if self.t.tv_nsec >= other.t.tv_nsec {
```

Comment:
Can you throw a quick comment in here explaining why this has to look exactly like it does, so this doesn't get "cleaned up" at some point in the future?